### PR TITLE
Improve avatar image loading

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

This PR ensures that we use a single instance of "coil.ImageLoader" (we don't want to rely on Coil instance bound to the application context as clients might rely on it as well) and  improves the performance of  `AvatarView`.

Notes:
- Custom fetcher is responsible for loading avatar bitmaps and creating unique cache keys for them
- `ImageLoader` in `stream-chat-android` is extended with a custom fetcher in `stream-chat-android-ui-components`
- Simplified API of `AvatarView` - `users` param is removed from `setChanelData(channel)` method
- Used "startup" library to automatically add the custom fetcher to `ImageLoader`
- Added temp solution for the problem when `User.isCurrentUser()` is used in the utility methods / unauthorized mode. Should be revisited later as this method is used everywhere which complicates testing. 

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

| Channel list | Component browser |
| --- | ---|
| avatar-2 | avatar-2
 |

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/startup/ImageLoaderFactoryInitializer.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import coil.ImageLoaderFactory
+import coil.decode.GifDecoder
+import coil.decode.ImageDecoderDecoder
+import com.getstream.sdk.chat.coil.StreamCoil
+import com.getstream.sdk.chat.coil.StreamImageLoaderFactory
+import io.getstream.chat.android.ui.avatar.AvatarBitmapFactory
+import io.getstream.chat.android.ui.coil.AvatarFetcher
+
+@SuppressWarnings("unused")
+public class ImageLoaderFactoryInitializer : Initializer<ImageLoaderFactory> {
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/startup/ImageLoaderFactoryInitializer.kt:15]: Is this required to be `public` by the startup library?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/coil/AvatarFetcher.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        val targetSize = size.let { if (it is PixelSize) it.width else 0 }
+        val resources = options.context.resources
+        return when (data) {
+            is Avatar.UserAvatar -> {
+                avatarBitmapFactory.createUserBitmap(
+                    data.user,
+                    data.avatarStyle,
+                    targetSize
+                ).let {
+                    DrawableResult(BitmapDrawable(resources, it), false, DataSource.MEMORY)
```
[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/coil/AvatarFetcher.kt:39]: Let me check if it affects something

[the remark this answers is not in the document]
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/coil/AvatarFetcher.kt:39]: Ok, might affect next steps in the pipeline (like crossfading). But I tried and I see no visual difference. So, leaving as is.

[the remark this answers is not in the document]
Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/coil/AvatarFetcher.kt:39]: Thanks for checking! Agreed, we can leave it like this then.

**Comment by bychkovdmitry** [comment]
Removed coil api calls from `AvatarView`

**Review by zsmb13 — DISMISSED** [review]
Nicely done!

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**